src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -43,6 +47,9 @@ def read(self, project: Project) -> RuleSet:
         return {rule: int(timestamp) for rule, timestamp in data.items()}
 
     def write(self, project: Project, rules: RuleSet) -> None:
+        if len(rules) == 0:
+            return
```

Comment:
This breaks an (unwritten) contract: Any call to `write` should clear the store (see line 57).
```suggestion
```